bolt/lib/Rewrite/DWARFRewriter.cpp

```diff
@@ -759,8 +751,14 @@ void DWARFRewriter::updateDebugInfo() {
       DebugRangesSectionWriter *TempRangesSectionWriter =
           CU->getVersion() >= 5 ? RangeListsWritersByCU[*DWOId].get()
                                 : LegacyRangesWritersByCU[*DWOId].get();
+      std::optional<std::string> DwarfOutputPath =
+          opts::DwarfOutputPath.empty()
+              ? std::nullopt
+              : std::optional<std::string>(opts::DwarfOutputPath.c_str());
+      std::string DWOName = DIEBlder.updateDWONameCompDir(
```

Comment:
Why does hoisting this out of the lambda mean you don't need the lock anymore?

Reply:
The lock was previously part of the old multithreading implementation which we weren't using. Previously the plan was to remove it in a followup PR but since we're moving it out of the lambda here it doesn't make sense to have the lock at all since DWOName is processed sequentially.